metabase/metabase
# Infer id numeric columns as non metrics

[Slack thread](https://metaboat.slack.com/archives/C064QMXEV9N/p1718761931598589)

### Description

This PR slightly tweaks the logic of detecting which columns are metric columns to exclude numeric ID-like columns as metrics. It will improve column auto selection on cartesian charts in cases when there are only numeric columns but some are IDs.

### How to verify

```sql
select distinct invoices.ACCOUNT_ID, count(*)
from invoices
group by invoices.ACCOUNT_ID
```

1. New -> SQL query -> use the query above
2. Run the query and switch the visualization type to `Bar`
3. It should autoselect account_id as dimension and count(*) as metric 

### Checklist
